static/app/views/dashboards/filtersBar.tsx

```diff
@@ -45,6 +53,18 @@ export default function FiltersBar({
 }: FiltersBarProps) {
   const {selection} = usePageFilters();
   const organization = useOrganization();
+  const currentUser = useUser();
+  const {teams: userTeams} = useUserTeams();
+  let hasEditAccess = true;
+  if (organization.features.includes('dashboards-edit-access')) {
+    hasEditAccess = checkUserHasEditAccess(
+      currentUser,
+      userTeams,
+      organization,
+      dashboardPermissions,
+      dashboardCreator
+    );
+  }
```

Comment:
Should we just pass around `hasEditAccess` from above? It seems like a simpler change than checking this at multiple locations.

Also, an idea for a future clean up is that I think we could put this into a context to store edit access information and create a hook called `useHasEditAccess` or something to avoid passing around the prop as much